server/server.go

```diff
@@ -118,6 +114,14 @@ func (r *rodeServer) BatchCreateOccurrences(ctx context.Context, occurrenceReque
 		return nil, createError(log, "error creating occurrences", err)
 	}
 
+	if err = r.resourceManager.BatchCreateGenericResources(ctx, occurrenceResponse.Occurrences); err != nil {
```

Comment:
Moving these calls after creating occurrences may make the error path a little tricker. For instance, if creating the generic resources fails with a 500, it wouldn't be unreasonable for a caller to try again (although none of our collectors do that at the moment). 

With the way the resource and version methods work, that's fine, but the occurrences would be duplicated.

Reply:
So the reason I ended up switching the order on this was because I wanted the create timestamps for generic resource versions to line up with the timestamps of the occurrence that produced them.  I was thinking that it might be desirable for the build occurrence for a generic resource and the resource itself to share the same timestamp, so filtering on that would be easier.  That being said, I'm not sure how important that actually is, and you're right about the error path here being more complicated as a result.

One way to handle this would to have rode add timestamps to occurrences with missing timestamps, [similar to what we do in grafeas-elasticsearch](https://github.com/rode/grafeas-elasticsearch/blob/1c94587f7ef0d121c27cc9c7610173ec883edcc4/go/v1beta1/storage/elasticsearch.go#L327-L329).  Then I could revert the ordering and still get the functionality I was looking for.

Thoughts?

Reply:
Created https://github.com/rode/rode/issues/84